pkg/sqs/queue_test.go

```diff
@@ -0,0 +1,88 @@
+package sqs_test
+
+import (
+	"context"
+	awsMocks "github.com/applike/gosoline/pkg/cloud/aws/mocks"
+	"github.com/applike/gosoline/pkg/mon/mocks"
+	gosoSqs "github.com/applike/gosoline/pkg/sqs"
+	sqsMocks "github.com/applike/gosoline/pkg/sqs/mocks"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+func TestRunQueueTestSuite(t *testing.T) {
+	suite.Run(t, new(queueTestSuite))
+}
+
+type queueTestSuite struct {
+	suite.Suite
+	executor *awsMocks.Executor
+	queue    gosoSqs.Queue
+}
+
+func (qs *queueTestSuite) SetupTest() {
+	logger := mocks.NewLoggerMockedAll()
+	sqsClient := new(sqsMocks.SQSAPI)
+	qs.executor = new(awsMocks.Executor)
+	qs.queue = gosoSqs.NewWithInterfaces(logger, sqsClient, qs.executor, &gosoSqs.Properties{
+		Url: "http://foo.bar.baz",
+	})
+}
+
+func (qs *queueTestSuite) TestSendBatch() {
+	msgs := make([]*gosoSqs.Message, 0)
+
+	// Test case 1: empty batch
+	err := qs.queue.SendBatch(context.Background(), msgs)
+	if !qs.Nil(err) {
+		qs.FailNow("empty messages slice should return nil and not call any other func")
+	}
+
+	// Test case 2: 1 message in batch, but triggers a request too long error
+	msgs = append(msgs, &gosoSqs.Message{
+		DelaySeconds:           aws.Int64(int64(1)),
+		MessageGroupId:         aws.String("group-1"),
+		MessageDeduplicationId: aws.String("hash-1"),
+		Body:                   aws.String("foo"),
+	})
+	awsErr := awserr.New(sqs.ErrCodeBatchRequestTooLong, "foo", nil)
+	qs.executor.
+		On("Execute", context.Background(), mock.AnythingOfType("aws.RequestFunction")).
+		Once().
+		Return(nil, awsErr)
+	err = qs.queue.SendBatch(context.Background(), msgs)
+	if !qs.Equal(awsErr, err) {
+		qs.FailNow("SendBatch should fail with only one message which triggers a request too long error")
+	}
+
+	// Test case 3: 3 messages in batch, but triggers a request too long error
```

Comment:
Can you please move the different est cases to different test methods? The queue is stateless, thus there shouldn't be any issue.
It's also slightly better structured :wink:

Reply:
please have a look at my freshly pushed code